src/DynamoAnalyzer/Helper/CsvHandler.cs

```diff
@@ -0,0 +1,151 @@
+using CsvHelper;
+using CsvHelper.Configuration;
+using CsvHelper.TypeConversion;
+using DynamoAnalyzer.Models;
+using Newtonsoft.Json;
+
+namespace DynamoAnalyzer.Helper
+{
+    /// <summary>
+    /// Provides methods to work with csv files
+    /// </summary>
+    public static class CsvHandler
+    {
+        private static CsvConfiguration csvConfiguration = new CsvConfiguration(CultureInfo.InvariantCulture)
+        {
+            NewLine = Environment.NewLine,
+        };
+
+        /// <summary>
+        /// Writes to a file the result of all the analyzed packages and it's DLLs
+        /// </summary>
+        /// <param name="packages"></param>
+        /// <param name="date"></param>
+        /// <returns></returns>
+        public static async Task<string> WritePackagesCsv(List<AnalyzedPackage> packages, DateTime date)
+        {
+            FileInfo file = new FileInfo(Path.Combine(AnalyzeEnvironment.GetWorkspace().FullName, $"results[{date:yyyy-MM-dd_HH-mm-ss}].csv"));
+
+            using (var writer = new StreamWriter(file.FullName))
+            using (var csv = new CsvWriter(writer, csvConfiguration))
+            {
+                csv.Context.RegisterClassMap<AnalyzedPackageMap>();
+                await csv.WriteRecordsAsync(packages);
+            }
+
+            return file.FullName;
+        }
+
+        /// <summary>
+        /// Writes to a file the number of times that a DLL name is referenced from other packages
+        /// </summary>
+        /// <param name="packages"></param>
+        /// <param name="date"></param>
+        /// <returns></returns>
+        public static async Task<string> WriteDuplicatedCsv(IEnumerable<DuplicatedPackage> packages, DateTime date)
+        {
+            FileInfo file = new FileInfo(Path.Combine(AnalyzeEnvironment.GetWorkspace().FullName, $"duplicated[{date:yyyy_MM_dd_HH_mm_ss}].csv"));
+
+            using (var writer = new StreamWriter(file.FullName))
+            using (var csv = new CsvWriter(writer, csvConfiguration))
+            {
+                csv.Context.RegisterClassMap<DuplicatedPackageMap>();
+                await csv.WriteRecordsAsync(packages);
+            }
+
+            return file.FullName;
+        }
+    }
+
+    /// <summary>
+    /// Defines the properties to be written to a file
+    /// </summary>
+    public class DuplicatedPackageMap : ClassMap<DuplicatedPackage>
+    {
+        public DuplicatedPackageMap()
+        {
+            Map(m => m.ArtifactName).Name(nameof(DuplicatedPackage.ArtifactName));
+            Map(m => m.Count).Name(nameof(DuplicatedPackage.Count));
+            Map(m => m.Packages).TypeConverter<ArrayConverter>();
+        }
+    }
+
+    /// <summary>
+    /// Defines the properties to be written to a file
+    /// </summary>
+    public class AnalyzedPackageMap : ClassMap<AnalyzedPackage>
+    {
+        public AnalyzedPackageMap()
+        {
+            Map(m => m.Index).Name(nameof(AnalyzedPackage.Index));
```

Comment:
do we really care about all these properties? 

IMO all I need to know is if the package of a particular version can be converted easily or not.

Reply:
The Index property is used to sort the dll references of the analyzed packages before writting the results file, I think we should keep them, so we have several ways to sort the data in the csv file